### PR TITLE
Migrate source-postgres-strict-encrypt logic into source-postgres and make it the default cloud image

### PR DESCRIPTION
**Pull request by akashkulk**
Steps taken :

1. Copy over logic from .*StrictEncrypt.java classes 
2. Update the images in the ported over java test files to use source-postgres instead of the strict encrypt variant
3. Copy over the expected_spec.json file
4. Migrate integration tests
5. Update CAT tests

Future steps :
1. Remove the source-postgres-strict-encrypt directory
2. Remove Dockerfile?

### CODE REVIEW

**Comment by airbyte-oss-build-runner** [comment]
##  source-postgres test report (commit [`82cd548048`](https://github.com/airbytehq/airbyte/pull/31486/commits/82cd5480483de21d1cc7a47ce5760a80081d936d)) - ✅

⏲️  Total pipeline duration: 22mn10s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Build connector tar                                             | ✅       |
| Java Connector Unit Tests                                       | ✅       |
| Build source-postgres docker image for platform(s) linux/x86_64 | ✅       |
| Acceptance tests                                                | ✅       |
| Java Connector Integration Tests                                | ✅       |
| Validate metadata for source-postgres                           | ✅       |
| Connector version semver check                                  | ✅       |
| Connector version increment check                               | ✅       |
| QA checks                                                       | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/akash-variant-remove_source-postgres-attempt-2/1697501550/82cd5480483de21d1cc7a47ce5760a80081d936d/source-postgres/3.1.14/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:82cd5480483de21d1cc7a47ce5760a80081d936d)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-postgres test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-postgres test report (commit [`aec3f12f8f`](https://github.com/airbytehq/airbyte/pull/31486/commits/aec3f12f8f6326479e359e0fbaf1b74a69a71b78)) - ✅

⏲️  Total pipeline duration: 21mn36s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Build connector tar                                             | ✅       |
| Java Connector Unit Tests                                       | ✅       |
| Build source-postgres docker image for platform(s) linux/x86_64 | ✅       |
| Acceptance tests                                                | ✅       |
| Java Connector Integration Tests                                | ✅       |
| Validate metadata for source-postgres                           | ✅       |
| Connector version semver check                                  | ✅       |
| Connector version increment check                               | ✅       |
| QA checks                                                       | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/akash-variant-remove_source-postgres-attempt-2/1697504689/aec3f12f8f6326479e359e0fbaf1b74a69a71b78/source-postgres/3.1.14/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:aec3f12f8f6326479e359e0fbaf1b74a69a71b78)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-postgres test
```

File: airbyte-integrations/connectors/source-postgres/metadata.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ data:
   connectorSubtype: database
   connectorType: source
   definitionId: decd338e-5647-4c0b-adf4-da0e75f5a750
-  dockerImageTag: 3.1.13
```
Comment (bnchrch) [inline airbyte-integrations/connectors/source-postgres/metadata.yaml:9]: 💅 This should be a minor instead of a patch

**Review by bnchrch — APPROVED** [review]
So excited to see this go up for review.

From a connector ops perspective it seems good!

Thanks @akashkulk 

[goforit](https://app.graphite.dev/macros/airbytehq)

File: airbyte-integrations/connectors/source-postgres/metadata.yaml (hunk 1 above)
Comment (postamar) [reply airbyte-integrations/connectors/source-postgres/metadata.yaml:9]: +1

File: airbyte-integrations/connectors/source-postgres/build.gradle [hunk 2]
```diff
@@ -9,7 +9,7 @@ plugins {
 airbyteJavaConnector {
     cdkVersionRequired = '0.1.9'
     features = ['db-sources']
-    useLocalCdk = false
+    useLocalCdk = true
```
Comment (postamar) [inline airbyte-integrations/connectors/source-postgres/build.gradle:12]: Don't forget to publish the CDK, import the new version and set this to `false` before merging.

File: airbyte-integrations/connectors/source-postgres/metadata.yaml (hunk 1 above)
Comment (postamar) [reply airbyte-integrations/connectors/source-postgres/metadata.yaml:9]: +1

File: airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java [hunk 3]
```diff
@@ -150,6 +161,22 @@ public static Source sshWrappedSource() {
     this.stateEmissionFrequency = INTERMEDIATE_STATE_EMISSION_FREQUENCY;
   }
 
+  private ConnectorSpecification modifySpec(final ConnectorSpecification originalSpec) {
+    final ConnectorSpecification spec = Jsons.clone(originalSpec);
+    final ObjectNode properties = (ObjectNode) spec.getConnectionSpecification().get("properties");
```
Comment (postamar) [inline airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java:166]: Dumb question but is there any chance that the `properties` field is unset in the json and this ObjectNode is null?

File: airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java [hunk 4]
```diff
@@ -150,6 +161,22 @@ public static Source sshWrappedSource() {
...
+    ((ObjectNode) properties.get(SSL_MODE)).put("default", SSL_MODE_REQUIRE);
+
+    return spec;
+  }
+
+  @Override
+  public ConnectorSpecification spec() throws Exception {
+    if (DEPLOYMENT_MODE != null && DEPLOYMENT_MODE.equalsIgnoreCase(AdaptiveSourceRunner.CLOUD_MODE)) {
```
Comment (postamar) [inline airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java:174]: FYI `if (AdaptiveSourceRunner.CLOUD_MODE.equalsIgnoreCase(DEPLOYMENT_MODE)) {` is the same thing but shorter. Same comment applies elsewhere in this diff.

File: airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java [hunk 5]
```diff
@@ -150,6 +161,22 @@ public static Source sshWrappedSource() {
...
+      return modifySpec(super.spec());
```
Comment (postamar) [inline airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java:175]: nit: you might as well inline `modifySpec` here

File: docs/integrations/sources/postgres.md [hunk 6]
```diff
@@ -291,6 +291,7 @@ According to Postgres [documentation](https://www.postgresql.org/docs/14/datatyp
 
 | Version | Date       | Pull Request                                             | Subject                                                                                                                                                                    |
 |---------|------------|----------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| 3.1.14  | 2023-10-16 | [31029](https://github.com/airbytehq/airbyte/pull/31029) | [Internal Refactor] : Remove strict-encrypt variants.                                                                                                                      |
```
Comment (postamar) [inline docs/integrations/sources/postgres.md:294]: Is it really the case that this is an internal refactor? Strictly speaking (ha!) this connector now behaves differently when the env var `DEPLOYMENT_MODE=CLOUD` is set.

**Review by postamar — COMMENTED** [review]
Just a bunch of nits on my end, no objections of principle.

File: airbyte-integrations/connectors/source-postgres/build.gradle (hunk 2 above)
Comment (akashkulk) [reply airbyte-integrations/connectors/source-postgres/build.gradle:12]: Ah yes! Just wanted to do that to make sure tests were passing!

File: airbyte-integrations/connectors/source-postgres/metadata.yaml (hunk 1 above)
Comment (akashkulk) [reply airbyte-integrations/connectors/source-postgres/metadata.yaml:9]: Done

File: airbyte-integrations/connectors/source-postgres/build.gradle (hunk 2 above)
Comment (akashkulk) [reply airbyte-integrations/connectors/source-postgres/build.gradle:12]: Am publishing it, making the switch after it's been published and works locally with the newly published version

**Review by bazarnov — APPROVED** [review]

**Comment by airbyte-oss-build-runner** [comment]
##  source-postgres test report (commit [`85a40ed5b0`](https://github.com/airbytehq/airbyte/pull/31486/commits/85a40ed5b081ef32a01303e1769867682ab4ec6c)) - ❌

⏲️  Total pipeline duration: 19mn41s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Build connector tar                                             | ✅       |
| Java Connector Unit Tests                                       | ❌       |
| Build source-postgres docker image for platform(s) linux/x86_64 | ✅       |
| Acceptance tests                                                | ✅       |
| Java Connector Integration Tests                                | ✅       |
| Validate metadata for source-postgres                           | ✅       |
| Connector version semver check                                  | ✅       |
| Connector version increment check                               | ✅       |
| QA checks                                                       | ❌       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/akash-variant-remove_source-postgres-attempt-2/1697649538/85a40ed5b081ef32a01303e1769867682ab4ec6c/source-postgres/3.2.13/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:85a40ed5b081ef32a01303e1769867682ab4ec6c)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-postgres test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-postgres test report (commit [`db2002cd5a`](https://github.com/airbytehq/airbyte/pull/31486/commits/db2002cd5a1ee565e0131d6cf2726527c69a50ec)) - ❌

⏲️  Total pipeline duration: 18mn51s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Build connector tar                                             | ✅       |
| Java Connector Unit Tests                                       | ❌       |
| Build source-postgres docker image for platform(s) linux/x86_64 | ✅       |
| Acceptance tests                                                | ✅       |
| Java Connector Integration Tests                                | ✅       |
| Validate metadata for source-postgres                           | ✅       |
| Connector version semver check                                  | ✅       |
| Connector version increment check                               | ✅       |
| QA checks                                                       | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/akash-variant-remove_source-postgres-attempt-2/1697653201/db2002cd5a1ee565e0131d6cf2726527c69a50ec/source-postgres/3.2.13/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:db2002cd5a1ee565e0131d6cf2726527c69a50ec)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-postgres test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-postgres test report (commit [`e156d376da`](https://github.com/airbytehq/airbyte/pull/31486/commits/e156d376daebba6cf36e5f904134012d2d98aac6)) - ✅

⏲️  Total pipeline duration: 18mn36s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Build connector tar                                             | ✅       |
| Java Connector Unit Tests                                       | ✅       |
| Build source-postgres docker image for platform(s) linux/x86_64 | ✅       |
| Acceptance tests                                                | ✅       |
| Java Connector Integration Tests                                | ✅       |
| Validate metadata for source-postgres                           | ✅       |
| Connector version semver check                                  | ✅       |
| Connector version increment check                               | ✅       |
| QA checks                                                       | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/akash-variant-remove_source-postgres-attempt-2/1697655118/e156d376daebba6cf36e5f904134012d2d98aac6/source-postgres/3.2.13/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:e156d376daebba6cf36e5f904134012d2d98aac6)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-postgres test
```

**Pull request merged**